### PR TITLE
Add HasChanged method

### PR DESCRIPTION
**Pull request by annelo-msft**
Described in issue #7440.

### CODE REVIEW

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs [hunk 1]
```diff
@@ -657,5 +657,86 @@ private Request CreateGetRevisionsRequest(SettingSelector selector, string pageL
...
+                }
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        private Request CreateGetHeadRequest(string key, string label)
```
Comment (pakrym) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs]:
```suggestion
        private Request CreateHeadRequest(string key, string label)
```

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient_private.cs [hunk 2]
```diff
@@ -36,13 +35,49 @@ public partial class ConfigurationClient
...
+            }
+            else
+            {
+                result = ConfigurationServiceSerializer.DeserializeSetting(response.ContentStream);
+            }
+
+            return new Response<ConfigurationSetting>(response, result);
+        }
+
+        private static ConfigurationSetting CreateSettingFromHeaders(Response response)
```
Comment (pakrym) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient_private.cs]: Not a fan of this pattern, just get the etag header value in `HasChanged` method itself

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSetting.cs [hunk 3]
```diff
@@ -16,8 +16,10 @@ public sealed class ConfigurationSetting : IEquatable<ConfigurationSetting>
     {
         private IDictionary<string, string> _tags;
 
-        // TODO (pri 3): this is just for deserialization. We can remove after we move to JsonDocument
-        internal ConfigurationSetting() { }
+        internal ConfigurationSetting()
+        {
+            ETag = new ETag(default);
```
Comment (pakrym) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationSetting.cs]: ETag is a struct, it is initialized with a value equal to `new ETag(default)` by default.

**Review by pakrym — APPROVED** [review]
Other than minor comments.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs [hunk 4]
```diff
@@ -657,5 +657,86 @@ private Request CreateGetRevisionsRequest(SettingSelector selector, string pageL
...
+        {
+            if (string.IsNullOrEmpty(key))
+                throw new ArgumentNullException($"{nameof(key)}");
```
Comment (heaths) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs]: An empty string shouldn't throw an `ArgumentNullException`. This isn't consistent with the .NET Framework (or Core). You can use the `Argument` shared-source class from Azure.Core for ease.

**Review by heaths — APPROVED** [review]

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs (hunk 4 above)
Comment (annelo-msft) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs]: Filing separate issue to track this: #7732

**Pull request merged**